src/olympia/users/models.py

```diff
@@ -864,13 +873,23 @@ def allow_request(cls, request):
             request.user.email))
 
 
-class DeveloperAgreementRestriction:
+class DeveloperAgreementRestriction(GetErrorMessageMixin):
     error_message = _('Before starting, please read and accept our Firefox'
                       ' Add-on Distribution Agreement as well as our Review'
                       ' Policies and Rules. The Firefox Add-on Distribution'
                       ' Agreement also links to our Privacy Notice which'
                       ' explains how we handle your information.')
 
+    @classmethod
+    def get_error_message(cls, is_api):
+        if is_api:
+            url = 'https://addons.mozilla.org/en-US/developers/addon/api/key/'
```

Comment:
Nit: let's not hardcode the URL.

```suggestion
            from olympia.amo.templatetags.jinja_helpers import absolutify
            url = absolutify(reverse('devhub.api_key'))
```
Requires to fix to `test_has_permission_user_has_not_read_agreement_when_using_API` test accordingly (tests run with the `http://testserver` URL prefix, so the URL will be `http://testserver/en-US/developers/addon/api/key/` in tests)